annotation/tests/test_annotations.py

```diff
@@ -1,85 +1,252 @@
 
 
-@pytest.mark.unittest
 @pytest.mark.parametrize(
-    [
-        "row",
-        "expected_dictionary",
-    ],
-    [
+    (
+        "s3_prefix",
+        "bucket_name",
+        "expected_string",
+    ),
+    (
         (
-            ANNOTATION_ROW_TABLE,
-            {
-                "uuid_attr": "34c665fd-ddfb-412c-a3f8-3351d87c6030",
-                "datetime_attr": "2024-01-01T10:10:00",
-                "str_attr": "test string",
-                "int_attr": 1,
-            },
-            # row_to_dict won't cast INTEGER to string
+            "S3_test",
+            "bucket_test",
+            "S3_test-bucket_test",
         ),
         (
-            ANNOTATION_ROW,
-            {
-                "uuid_attr": "34c665fd-ddfb-412c-a3f8-3351d87c6030",
-                "datetime_attr": "2024-01-01T10:10:00",
-                "str_attr": "test string",
-                "int_attr": 1,
-            },
-            # it return the same as the previous case
+            None,
+            "bucket_test",
+            "bucket_test",
         ),
-    ],
+    ),
 )
-def test_row_to_dict(row, expected_dictionary):
-    result = row_to_dict(row)
-    print(result)
-    assert result == expected_dictionary
+def test_convert_bucket_name_if_s3prefix(
+    s3_prefix: string,
+    bucket_name: string,
+    expected_string: string,
+):
+    with patch("annotation.annotations.main.S3_PREFIX", s3_prefix):
+        result = convert_bucket_name_if_s3prefix(bucket_name=bucket_name)
+        assert result == expected_string
+
+
+@pytest.mark.parametrize(
+    ("s3_provider",),
+    (
+        ("minio",),
+        ("aws_iam",),
+    ),
+)
+def test_connect_s3(moto_s3: boto3.resource, s3_provider: string):
+    with patch("boto3.resource", return_value=moto_s3) as mock_resource, patch(
+        "annotation.annotations.main.S3_PROVIDER", s3_provider
+    ):
+        result_s3 = connect_s3(TEST_TENANT)
+        mock_resource.assert_called_once()
+        assert result_s3 == moto_s3
+
+
+def test_connect_s3_no_provider(moto_s3: boto3.resource):
+    with patch("boto3.resource", return_value=moto_s3), patch(
+        "annotation.annotations.main.S3_PROVIDER", "NO_PROVIDER"
+    ):
+        with pytest.raises(NotConfiguredException):
+            connect_s3(TEST_TENANT)
+
+
+def test_connect_s3_no_bucket(moto_s3: boto3.resource):
+    with patch("boto3.resource", return_value=moto_s3), patch(
+        "annotation.annotations.main.S3_PROVIDER", "aws_iam"
+    ):
+        with pytest.raises(moto_s3.meta.client.exceptions.NoSuchBucket):
+            connect_s3("NO_BUCKET")
+
+
+def test_get_sha_of_bytes():
+    assert sha1(b"1").hexdigest() == get_sha_of_bytes(b"1")
+
+
+def test_create_manifest_json(
+    moto_s3: boto3.resource, annotated_doc, annotation_manifest
+):
+    db = Mock(spec=Session)
+    db.query().filter().order_by().all.return_value = []
+    s3_path = (
+        f"annotation/{str(annotated_doc.job_id)}/{str(annotated_doc.file_id)}"
```

Comment:
f string already call `__str__` method on every interpolated value, no need for such explicit type case.